draft-ietf-quic-http.md

```diff
@@ -591,6 +591,83 @@ The payload consists of:
   : HPACK-compressed request headers for the promised response.
 
 
+### GOAWAY {#frame-goaway}
+
+The GOAWAY frame (type=0x7) is used to initiate graceful shutdown of a
+connection by a server.  GOAWAY allows a server to stop accepting new requests
+while still finishing processing of previously received requests.  This enables
+administrative actions, like server maintenance.  GOAWAY by itself does not
+close a connection.
+
+The GOAWAY frame does not define any flags, and the payload is a QUIC stream
+identifier.  The GOAWAY frame applies to the connection, not a specific stream.
+An endpoint MUST treat a GOAWAY frame on a stream other than the control stream
+as a connection error ({{errors}}) of type HTTP_WRONG_STREAM.
+
+There is an inherent race condition between a client sending new requests and
+the server sending a GOAWAY frame.  To deal with this case, the GOAWAY contains
+the stream identifier of the last client-initiated request that was or might be
+processed in this connection.  This MAY be lower than the stream limit
+identified by a QUIC MAX_STREAM_ID frame, and MAY be zero if no requsets were
+processed.  Servers SHOULD NOT increase the MAX_STREAM_ID limit after sending a
+GOAWAY frame.
+
+  Note:
+  : In this context, "processed" means that some data from the stream was
+    passed to some higher layer of software that might have taken some action as
+    a result.
+
+Once sent, the server will refuse requests sent on streams with an identifier
+higher than the included last stream identifier.  Clients MUST NOT send new
+requests on the connection after receiving GOAWAY, although requests might
+already be in transit. A new connection can be established for new requests.
+
+If the client has sent requests on streams with a higher stream identifier than
+indicated in the GOAWAY frame, those requests were not and will not be
+processed.  Servers SHOULD reset new streams above this ID with the error code
+HTTP_REQUEST_CANCELLED.  Servers MAY also reset streams below the indicated ID
```

Comment:
s/Servers SHOULD reset new streams/Endpoints SHOULD reset client-initiated streams/

It's not especially graceful, but the server might see those streams as open, and cancelling them avoids problems.  Also, this isn't about the server, because clients should also reset their requests and avoid retransmissions.  


Separately, I wonder: HTTP_REQUEST_CANCELLED from the client doesn't have the effect that we want: it doesn't guarantee that the request wasn't processed by the server; it can't.  Do we need some text on that.